sphinx/directives/code.py

```diff
@@ -53,8 +53,15 @@ def run(self):
         if 'linenothreshold' in self.options:
             try:
                 linenothreshold = int(self.options['linenothreshold'])
-            except Exception:
-                linenothreshold = 10
+            except ValueError:
+                default_linenothreshold = 10
+                location = self.state_machine.get_source_and_line(self.lineno)
+                logger.warning(__('linenothreshold value must be int typed. '
+                                  'Passed %r. Using default value (%d).'
+                                  % (self.options['linenothreshold'],
+                                     default_linenothreshold)),
+                              location=location)
```

Comment:
It seems flake8 warns this line. Check it please.
https://travis-ci.org/sphinx-doc/sphinx/jobs/373627298#L491